i3lock.c

```diff
@@ -169,29 +174,6 @@ static bool load_compose_table(const char *locale) {
     return true;
 }
 
-/*
- * Clears the memory which stored the password to be a bit safer against
- * cold-boot attacks.
- *
- */
-static void clear_password_memory(void) {
-#ifdef __OpenBSD__
-    /* Use explicit_bzero(3) which was explicitly designed not to be
-     * optimized out by the compiler. */
-    explicit_bzero(password, strlen(password));
-#else
-    /* A volatile pointer to the password buffer to prevent the compiler from
-     * optimizing this out. */
-    volatile char *vpassword = password;
-    for (size_t c = 0; c < sizeof(password); c++)
-        /* We store a non-random pattern which consists of the (irrelevant)
-         * index plus (!) the value of the beep variable. This prevents the
-         * compiler from optimizing the calls away, since the value of 'beep'
-         * is not known at compile-time. */
-        vpassword[c] = c + (int)beep;
-#endif
-}
-
```

Comment:
I might be missing something here but aren't you supposed to call `explicit_bzero()` here?

Reply:
The call moved into the define above: `#define clear_password_memory() explicit_bzero(password, strlen(password))`

Reply:
Oh I see, I didn't notice we were in a function.